test/unit/ar_id_map_test.rb

```diff
@@ -1,26 +1,17 @@
-require "test_helper"
+if defined?(ActiveRecord::IdentityMap) && ActiveRecord::IdentityMap.respond_to?(:without)
+  require "mocha"
+  require "test_helper"
 
-class ARIdMapTest < ActiveSupport::TestCase
-  setup do
-    @widget = Widget.new
-    @widget.update_attributes name: "Henry", created_at: Time.now - 1.day
-    @widget.update_attributes name: "Harry"
-  end
+  class ARIdMapTest < ActiveSupport::TestCase
+    setup do
+      @widget = Widget.new
+      @widget.update_attributes name: "Henry", created_at: Time.now - 1.day
+      @widget.update_attributes name: "Harry"
+    end
 
-  if defined?(ActiveRecord::IdentityMap) && ActiveRecord::IdentityMap.respond_to?(:without)
     should "not clobber the IdentityMap when reifying" do
-      module ActiveRecord::IdentityMap
-        class << self
-          alias __without without
-          def without(&block)
-            @unclobbered = true
-            __without(&block)
-          end
-        end
-      end
-
+      ActiveRecord::IdentityMap.expects(:without).once
```

Comment:
From what I can understand from https://github.com/airblade/paper_trail/commit/5df9949ff41f22b2588915d87deff59ee4ad3974, all this was previously testing for, was that `ActiveRecord::IdentityMap.without` was called. That seems like the perfect use case for a mock.

I prefer avoiding dependencies, but IMO adding a dependency is the lesser evil here compared to the roundabout way of testing this via monkeypatching AR in the test.

The only test in this file only gets run when `ActiveRecord::IdentityMap` is defined, which is to say only for AR3, so I moved the `defined?` check to the top level of the file, so that we can add the new mocha dependency to the AR3 gemfile only.